src/imodmodel/parsers.py

```diff
@@ -56,12 +73,8 @@ def _parse_object_header(file: BinaryIO) -> ObjectHeader:
 
 
 def _parse_object(file: BinaryIO) -> Object:
-    header = _parse_object_header(file)
-    contours = []
-    for _ in range(header.contsize):
-        _parse_control_sequence(file)
-        contours.append(_parse_contour(file))
-    return Object(contours=contours)
+    _parse_object_header(file)
+    return Object()
```

Comment:
am I reading this right that currently you are not returning a valid object here? i.e. this breaks current parsing

Reply:
Yes, I changed the code such that the object doesn't "parse itself" anymore. I found it easier to determine the next chunk ID in parse_model() and call the appropriate parsing function there. Otherwise, if _parse_object() runs into the end of its description in the file, it has already consumed the next chunk ID (let's say it's another object). When it returns to the caller, the caller doesn't know what the next chunk ID is. I was at first playing with moving the file pointer back by 4 bytes, but that felt hacky and I'm not sure if this works everywhere.

 _parse_object_header and _parse_object are pretty much equivalent right now, and the returned Object() is empty which I now realize is not so nice. My solution would be to merge _parse_object() and _parse_object_header() and also merge Object and ObjectHeader.

Before, you were using header.contsize to parse the appropriate number of contours. I would rather just parse every contour, mesh and whatever else there is and then, when the parsing is done, validate that header.contsize agrees with the actual number of contours and throw a warning otherwise.

Reply:
This sounds cleaner and I'm open to unification/cleanup, thanks for taking the time to engage with my code - that work can be in a separate PR before or after this one if you want to keep this simpler 🙂